src/Job/DeveloperSync.php

```diff
@@ -130,21 +131,42 @@ protected function executeRequest() {
   /**
    * {@inheritdoc}
    */
-  public function execute() : bool {
+  public function execute(): bool {
     parent::execute();
 
+    // Update Apigee Edge developers and Drupal users if needed.
+    $identical_accounts = array_intersect($this->edgeAccounts, $this->drupalAccounts);
+    foreach ($identical_accounts as $search => $mail) {
+      /** @var \Apigee\Edge\Api\Management\Entity\DeveloperInterface $developer */
+      $developer = Developer::load($mail);
+      /** @var \Drupal\user\UserInterface $account */
+      $account = user_load_by_mail($mail);
```

Comment:
The same of users objects, fetch their IDs by mail and then load all of them before this loop.